File: test/python/kms_test.py
import pytest
from python_terratest import Terraform
import os

def test_kms_module_versions():
    # Get all version directories
    versions_dir = '../modules/kms'
    version_dirs = [d for d in os.listdir(versions_dir) if os.path.isdir(os.path.join(versions_dir, d))]
    
    for version in version_dirs:
        with pytest.subTest(version=version):
            # Configure Terraform options for this version
            terraform_options = {
                'terraform_dir': f'../modules/kms/{version}',
                'vars': {},
                'env_vars': {
                    'AWS_ACCESS_KEY_ID': 'test',
                    'AWS_SECRET_ACCESS_KEY': 'test',
                    'AWS_REGION': 'us-east-1'
                }
            }

            # Initialize Terraform
            tf = Terraform(**terraform_options)
            
            try:
                # Initialize and apply
                tf.init()
                tf.apply()
                
                # Get output
                output = tf.output('example_output')
                assert output is not None
                assert output != ''
                
            finally:
                # Cleanup
                tf.destroy() 

File: test/go/kms_test.go
package test

import (
	"os"
	"path/filepath"
	"testing"

	"github.com/gruntwork-io/terratest/modules/terraform"
)

func TestKmsModule(t *testing.T) {

	versionsDir := "../../modules/kms"
	versionDirs, err := os.ReadDir(versionsDir)
	if err != nil {
		t.Fatal(err)
	}

	for _, version := range versionDirs {
		if version.IsDir() {
			t.Run(version.Name(), func(t *testing.T) {
				terraformOptions := &terraform.Options{
					TerraformDir: filepath.Join(versionsDir, version.Name()),
					Vars:         map[string]interface{}{},
					EnvVars: map[string]string{
						"AWS_ACCESS_KEY_ID":     "test",
						"AWS_SECRET_ACCESS_KEY": "test",
						"AWS_REGION":            "us-east-1",
					},
				}

				// Initialize and validate terraform configuration
				terraform.Init(t, terraformOptions)

				// Validate the terraform configuration
				terraform.Validate(t, terraformOptions)

				// Since this is a module test, we mainly want to validate syntax
				// Full integration tests should be done with actual infrastructure
				t.Logf("KMS module %s validation completed successfully", version.Name())
			})
		}
	}
}
